pkg/ddl/column.go

```diff
@@ -610,10 +618,46 @@ func (w *worker) updateCurrentElement(
 		if err != nil {
 			return errors.Trace(err)
 		}
+		if err = prepareForAddNextIndex(reorgInfo, rh); err != nil {
+			return errors.Trace(err)
+		}
 	}
 	return nil
 }
 
+func prepareForAddNextIndex(reorgInfo *reorgInfo, rh *reorgHandler) error {
+	if reorgInfo.ReorgMeta.IsFastReorg && !reorgInfo.ReorgMeta.IsDistReorg {
+		return rh.ResetCheckpoint(reorgInfo.Job)
+	}
+	return nil
+}
+
+func (w *worker) resetReorgMeta(reorgMeta *model.DDLReorgMeta, job *model.Job) func() {
```

Comment:
If we use DXF to add index, the concurrency and the max node count are calculated automatically (see https://github.com/pingcap/tidb/pull/62995). Remember to adapt the strategy before merging this.